File: Code Chef/JavaScript/700-799/718 Speed Limit Test SPEEDTEST.js
let input = '';
process.stdin.on('data', (data) => input += data);

const strToInt = (strArray) => {
    strArray.shift();
    const intArray = strArray.map((strItem) => strItem.trim().split(' ').map(Number));
    
    return intArray;
};

const whoDrivesFaster = (distanceForAlice, timeUsedByAlice, distanceForBob, timeUsedByBob) => {
    const speedOfAlice = distanceForAlice / timeUsedByAlice;
    const speedOfBob = distanceForBob / timeUsedByBob;
    
    (speedOfAlice === speedOfBob)
        ? console.log('Equal')
        : (speedOfAlice > speedOfBob)
            ? console.log('Alice')
            : console.log('Bob');
};

process.stdin.on('end', () => {
    const testcases = strToInt(input.trim().split('\n'));
    
    testcases.forEach((testcase) => {
        const [distanceForAlice, timeUsedByAlice, distanceForBob, timeUsedByBob] = testcase;
        whoDrivesFaster(distanceForAlice, timeUsedByAlice, distanceForBob, timeUsedByBob);
    });
});

File: Code Chef/C Programming/700-799/718 Speed Limit Test SPEEDTEST.c
#include <stdio.h>
int main(void)
{
	int t;
	float a,b,x,y;
	scanf("%d",&t);
	for(int i=0;i<t;i++)
	{
	    scanf("%f%f%f%f",&a,&x,&b,&y);
	    float sp1=a/x, sp2=b/y;
	    if(sp1==sp2)
	    {
	        printf("Equal\n");
	    }
	    else if(sp1<sp2)
	    {
	        printf("Bob\n");
	    }
	    else if(sp1>sp2)
	    {
	        printf("Alice\n");
	    }
	}
	return 0;
}
